spec/models/user_spec.rb

```diff
@@ -9,13 +9,39 @@
     expect(@user).to be_valid
   end
 
+  it "is invalid with a status other than 'available' or 'working'" do
+    @user.current_status = "unavailable"
+    expect(@user).not_to be_valid
+  end
+
+  it "is not valid with an email without @" do
+    @user.email = "testATexample.de"
+    expect(@user).not_to be_valid
+  end
+
   it "is not valid without an email" do
     @user.email = ""
     expect(@user).to_not be_valid
   end
 
   it "is not valid without a password" do
     @user.password = ""
+    @user.password_confirmation = ""
     expect(@user).not_to be_valid
   end
+
+  it "is not valid without a username" do
+    @user.username = ""
+    expect(@user).not_to be_valid
+  end
+
+  it "creates user name from email" do
+    user = described_class.new({ email: "test-user@example.org" })
```

Comment:
Or use a factory and specify the `email` attribute? https://github.com/hpi-swt2/connections-portal/blob/dev/spec/factories/users.rb